src/Microsoft.AspNetCore.Hosting.Server.Abstractions/Features/IServerAddressesFeature.cs

```diff
@@ -8,5 +8,7 @@ namespace Microsoft.AspNetCore.Hosting.Server.Features
     public interface IServerAddressesFeature
     {
         ICollection<string> Addresses { get; }
+
+        bool PreferHostingUrls { get; set; }
```

Comment:
Actually will try to make this an internal set on the concrete type and remove it from the interface. See if that works